src/components/forms/ProxyDeviceForm.tsx

```diff
@@ -0,0 +1,303 @@
+import { FC } from "react";
+import {
+  Button,
+  Icon,
+  Input,
+  Label,
+  useNotify,
+} from "@canonical/react-components";
+import { useQuery } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+import { LxdDeviceValue, LxdProxyDevice } from "types/device";
+import { InstanceAndProfileFormikProps } from "./instanceAndProfileFormValues";
+import { fetchProfiles } from "api/profiles";
+import { getInheritedProxies } from "util/configInheritance";
+import Loader from "components/Loader";
+import { ProxyCard } from "types/resources";
+import ScrollableForm from "components/ScrollableForm";
+import RenameDeviceInput from "components/forms/RenameDeviceInput";
+import ConfigurationTable from "components/ConfigurationTable";
+import { MainTableRow } from "@canonical/react-components/dist/components/MainTable/MainTable";
+import { getConfigurationRowBase } from "components/ConfigurationRow";
+import classnames from "classnames";
+import {
+  addNoneDevice,
+  deduplicateName,
+  findNoneDeviceIndex,
+  removeDevice,
+} from "util/formDevices";
+import { getInheritedDeviceRow } from "components/forms/InheritedDeviceRow";
+import { deviceKeyToLabel } from "util/devices";
+import { ensureEditMode } from "util/instanceEdit";
+import SelectProxyBtn from "components/forms/SelectProxyBtn";
+// import ProxyDeviceInput from "./ProxyDeviceInput";
+
+interface Props {
+  formik: InstanceAndProfileFormikProps;
+  project: string;
+}
+
+const ProxyDeviceForm: FC<Props> = ({ formik, project }) => {
+  const notify = useNotify();
+
+  const {
+    data: profiles = [],
+    isLoading: isProfileLoading,
+    error: profileError,
+  } = useQuery({
+    queryKey: [queryKeys.profiles],
+    queryFn: () => fetchProfiles(project),
+  });
+
+  if (profileError) {
+    notify.failure("Loading profiles failed", profileError);
+  }
+
+  const inheritedProxies = getInheritedProxies(formik.values, profiles);
+
+  const existingDeviceNames: string[] = [];
+  existingDeviceNames.push(...inheritedProxies.map((item) => item.key));
+  existingDeviceNames.push(...formik.values.devices.map((item) => item.name));
```

Comment:
I just realize now this approach is not 100% correct, because all device types share the same namespace. So we could make the `getExistingDeviceNames(formik)` function generic for all device pages. But we can add a follow-up task to do this. Probably best not to mix this change into this pr.

Reply:
Sorry, are you saying one should create the "getExistingDeviceNames" function? How would this work?

Reply:
We can add that in another PR, I'll log a ticket for it. Just realized we should do that when reading this PR.